narayana-spring-boot-starter/src/test/java/me/snowdrop/boot/narayana/autoconfigure/NarayanaConfigurationIT.java

```diff
@@ -99,4 +101,33 @@ public void pooledXaDataSourceWrapperShouldBeLoaded() {
         assertThat(xaDataSourceWrapper).isInstanceOf(PooledXADataSourceWrapper.class);
     }
 
+    @Test
+    public void genericXaConnectionFactoryWrapperShouldBeLoaded() {
+        Properties properties = new Properties();
+        properties.put("narayana.messaginghub.enabled", "false");
+        PropertiesPropertySource propertySource = new PropertiesPropertySource("test", properties);
+
+        this.context = new AnnotationConfigApplicationContext();
+        this.context.register(NarayanaConfiguration.class);
+        this.context.getEnvironment().getPropertySources().addFirst(propertySource);
+        this.context.refresh();
+
+        XAConnectionFactoryWrapper xaConnectionFactoryWrapper = this.context.getBean(XAConnectionFactoryWrapper.class);
+        assertThat(xaConnectionFactoryWrapper).isInstanceOf(GenericXAConnectionFactoryWrapper.class);
+    }
+
+    @Test
+    public void pooledXaConnectionFactoryWrapperShouldBeLoaded() {
```

Comment:
Could you also add one test to verify that the deprecated properties still work?

Reply:
Sure, actually none of them is really tested. Will add a bunch of test cases.